vpr/src/place/place_constraints.cpp

```diff
@@ -180,3 +181,28 @@ void load_cluster_constraints() {
         }
     }
 }
+
+void mark_fixed_blocks() {
+    auto& cluster_ctx = g_vpr_ctx.clustering();
+    auto& place_ctx = g_vpr_ctx.mutable_placement();
+    auto& floorplanning_ctx = g_vpr_ctx.floorplanning();
+
+    for (auto blk_id : cluster_ctx.clb_nlist.blocks()) {
+        PartitionRegion pr = floorplanning_ctx.cluster_constraints[blk_id];
+        std::vector<Region> part_region = pr.get_partition_region();
+        if (part_region.size() == 1) {
+            int subtile = part_region[0].get_sub_tile();
+            vtr::Rect<int> rect = part_region[0].get_region_rect();
+
+            if (part_region[0].locked()) {
```

Comment:
get rid of .locked() once you add the utility function because then it is no longer needed.